src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -313,6 +314,47 @@ describe('WorkspacesUsecases', () => {
     });
   });
 
+  describe('editWorkspaceDetails', () => {
+    const user = newUser();
+    const workspace = newWorkspace({ owner: user });
+    const editWorkspaceDto: EditWorkspaceDetailsDto = {
+      name: 'Test Workspace',
+      description: 'Workspace description',
+    };
+    it('When workspace does not exist, then it should throw', async () => {
+      jest.spyOn(workspaceRepository, 'findById').mockResolvedValueOnce(null);
+
+      await expect(
+        service.editWorkspaceDetails(workspace.id, user, editWorkspaceDto),
+      ).rejects.toThrow(NotFoundException);
+    });
+
+    it('When user is not the owner of the workspace, then it should throw', async () => {
+      jest
+        .spyOn(workspaceRepository, 'findById')
+        .mockResolvedValueOnce(workspace);
+
+      await expect(
+        service.editWorkspaceDetails(workspace.id, newUser(), editWorkspaceDto),
+      ).rejects.toThrow(ForbiddenException);
+    });
+
+    it('When the user is the owner of the workspace, then it should update the workspace details', async () => {
+      jest
+        .spyOn(workspaceRepository, 'findById')
+        .mockResolvedValueOnce(workspace);
+
+      await service.editWorkspaceDetails(workspace.id, user, editWorkspaceDto);
+
+      expect(workspaceRepository.updateBy).toHaveBeenCalledWith(
```

Comment:
Same here, you should assert that the method resolves with the expected returned content rather than leaving the test to crash